t/24-worker-engine.t

```diff
@@ -93,4 +95,36 @@ subtest 'asset settings' => sub {
 };
 
 
+subtest 'caching' => sub {
+    is(OpenQA::Worker::Engines::isotovideo::cache_assets, undef, 'cache_assets has nothing to do without assets');
+    my %assets = (
+        ISO => 'foo.iso',
+    );
+    my $cache_client = Test::MockModule->new('OpenQA::Worker::Cache::Client');
+    $cache_client->mock(availability_error => undef);
+    #$cache_client->mock(asset_path => undef);
+    $cache_client->mock(asset_exists => undef);
+    my $mojo_file = Test::MockModule->new('Mojo::File');
+    $mojo_file->mock(path => undef);
+    my $cache_request = Test::MockModule->new('OpenQA::Worker::Cache::Request');
+    $cache_request->mock(enqueue => undef);
+    $cache_request->mock(asset => sub { return OpenQA::Worker::Cache::Request->new });
+    $got = OpenQA::Worker::Engines::isotovideo::cache_assets(undef, $settings, \%assets);
+    like($got->{error}, qr/Failed to download/, 'cache_assets can not pick up supplied assets when not found') or diag explain $got;
+};
+
+$got = OpenQA::Worker::Engines::isotovideo::do_asset_caching($settings);
```

Comment:
Can you make this a subtest, too? And I miss the place where `$got` is declared/defined. I only see `my $got` in nested scopes.